apps/src/templates/teacherDashboard/AddSectionDialog.jsx

```diff
@@ -12,86 +12,109 @@ import {
   beginImportRosterFlow,
   setRosterProvider,
   editSectionProperties,
-  cancelEditingSection
+  cancelEditingSection,
+  assignedCourseOffering
 } from './teacherSectionsRedux';
 import ParticipantTypePicker from './ParticipantTypePicker';
+import Spinner from '@cdo/apps/code-studio/pd/components/spinner';
 
 /**
  * UI for a teacher to add a new class section.  For editing a section see
  * EditSectionDialog.
  */
-class AddSectionDialog extends Component {
-  static propTypes = {
-    // Provided by Redux
-    isOpen: PropTypes.bool.isRequired,
-    section: sectionShape,
-    beginImportRosterFlow: PropTypes.func.isRequired,
-    setRosterProvider: PropTypes.func.isRequired,
-    setLoginType: PropTypes.func.isRequired,
-    setParticipantType: PropTypes.func.isRequired,
-    handleCancel: PropTypes.func.isRequired,
-    availableParticipantTypes: PropTypes.arrayOf(PropTypes.string).isRequired
+const AddSectionDialog = ({
```

Comment:
I turned this component into a functional component while I was here. It didn't store any state so it was an easy change.

Reply:
I realize i have no idea when we should use const vs function() when translating to a functional component! Do we have a stated preference somewhere? Also THANKYOU for doing this functional translation at the same time!!

Reply:
not that I know of and I'm definitely guilty of being inconsistent here 🙈